### PR TITLE
2597 add bool fields api

### PR DESCRIPTION
**Pull request by purvinptl**
This PR closes https://github.com/GSA-TTS/FAC/issues/2597 

This branch includes:

bumped api versions 1_0_2 to 1_0_3

adds gen.is_additional_ueis, is_secondary_auditors, is_multiple_eins to /general

We did a full submission with the following parameters:

- is_multiple_eins set to No
- is_secondary_auditors set to No
- is_additional_ueis set to No

The API response correctly reflects the selection made, confirming that the request was successful.

We also conducted a full submission with the parameters reversed:

- is_multiple_eins set to Yes
- is_secondary_auditors set to Yes
- is_additional_ueis set to Yes

Once again, the API returned the correct selection, confirming the success of the request."

## PR checklist: submitters

- [x]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [x]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [x]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [ ]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [x]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [x]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [ ]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

File: backend/dissemination/api/api_v1_0_3/create_views.sql [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+        WHEN aud.general_information ->> 'multiple_eins_covered' = 'true' THEN 'Yes'
+        WHEN aud.general_information ->> 'multiple_eins_covered' = 'false' THEN 'No'
+        ELSE 'null'
+    END AS is_multiple_eins,
+    CASE
+        WHEN aud.general_information ->> 'secondary_auditors_exist' = 'true' THEN 'Yes'
+        WHEN aud.general_information ->> 'secondary_auditors_exist' = 'false' THEN 'No'
+        ELSE 'null'
+    END AS is_secondary_auditors
+    from
```
Comment (phildominguez-gsa) [inline backend/dissemination/api/api_v1_0_3/create_views.sql:268]: I haven't done cases before, but just checking on the indentation. I would have expected something like this:

```suggestion
        gen.data_source,
        gen.is_aicpa_audit_guide_included,
        gen.is_additional_ueis,
        CASE
            WHEN aud.general_information ->> 'multiple_eins_covered' = 'true' THEN 'Yes'
            WHEN aud.general_information ->> 'multiple_eins_covered' = 'false' THEN 'No'
            ELSE 'null'
        END AS is_multiple_eins,
        CASE
            WHEN aud.general_information ->> 'secondary_auditors_exist' = 'true' THEN 'Yes'
            WHEN aud.general_information ->> 'secondary_auditors_exist' = 'false' THEN 'No'
            ELSE 'null'
        END AS is_secondary_auditors
    from
```

File: backend/dissemination/api/api_v1_0_3/create_views.sql [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+        gen.total_amount_expended,
+        gen.type_audit_code,
+        gen.is_public,
+        gen.data_source,
+        gen.is_aicpa_audit_guide_included,
+        gen.is_additional_ueis,
+        CASE
+        WHEN aud.general_information ->> 'multiple_eins_covered' = 'true' THEN 'Yes'
+        WHEN aud.general_information ->> 'multiple_eins_covered' = 'false' THEN 'No'
+        ELSE 'null'
```
Comment (phildominguez-gsa) [inline backend/dissemination/api/api_v1_0_3/create_views.sql:261]: Is null possible for these fields?

**Review by phildominguez-gsa — CHANGES_REQUESTED** [review]

File: backend/dissemination/api/api_v1_0_3/create_views.sql (hunk 2 above)
Comment (eazapata23) [reply backend/dissemination/api/api_v1_0_3/create_views.sql:261]: I thought we were stepping away from having null values and having in place a N/A?

Comment (purvinptl) [reply backend/dissemination/api/api_v1_0_3/create_views.sql:261]: I can return the value as it is also.

CASE
    WHEN aud.general_information ->> 'multiple_eins_covered' = 'true' THEN 'Yes'
    WHEN aud.general_information ->> 'multiple_eins_covered' = 'false' THEN 'No'
    ELSE null
END AS is_multiple_eins,
CASE
    WHEN aud.general_information ->> 'secondary_auditors_exist' = 'true' THEN 'Yes'
    WHEN aud.general_information ->> 'secondary_auditors_exist' = 'false' THEN 'No'
    ELSE null
END AS secondary_auditors_exist

File: backend/audit/migrations/0005_alter_submissionevent_event.py
Comment (jadudm) [inline backend/audit/migrations/0005_alter_submissionevent_event.py:1]: Should there be a migration in this PR?

Comment (phildominguez-gsa) [reply backend/audit/migrations/0005_alter_submissionevent_event.py:1]: Nope, [view change only](https://gsa-tts.slack.com/archives/C03QCP4FSN8/p1698160608038229?thread_ts=1698093352.008659&cid=C03QCP4FSN8)

Comment (phildominguez-gsa) [reply backend/audit/migrations/0005_alter_submissionevent_event.py:1]: That was supposed to link to your slack msg about avoiding a migration but I'm not sure it works 😃

File: backend/dissemination/api/api_v1_0_3/create_views.sql (hunk 1 above)
Comment (purvinptl) [reply backend/dissemination/api/api_v1_0_3/create_views.sql]: Made the changes in latest commit.

File: backend/dissemination/api/api_v1_0_3/create_views.sql (hunk 2 above)
Comment (purvinptl) [reply backend/dissemination/api/api_v1_0_3/create_views.sql]: Made the changes to omit the Else statement in latest commit.

File: backend/audit/migrations/0005_alter_submissionevent_event.py
Comment (purvinptl) [reply backend/audit/migrations/0005_alter_submissionevent_event.py:1]: Resolved.

**Review by purvinptl — COMMENTED** [review]
Resolved the conflicts, indentation error, and null value.

**Review by phildominguez-gsa — APPROVED** [review]

**Review by danswick — APPROVED** [review]

File: backend/dissemination/api/api_v1_0_3/create_views.sql [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+        gen.number_months,
+        gen.audit_period_covered,
+        gen.total_amount_expended,
+        gen.type_audit_code,
+        gen.is_public,
+        gen.data_source,
+        gen.is_aicpa_audit_guide_included,
+        gen.is_additional_ueis,
+        CASE
+            WHEN aud.general_information ->> 'multiple_eins_covered' = 'true' THEN 'Yes'
```
Comment (jadudm) [inline backend/dissemination/api/api_v1_0_3/create_views.sql:259]:
```suggestion
            WHEN aud.general_information ->> 'multiple_eins_covered' = 'true' THEN 'Y'
```

File: backend/dissemination/api/api_v1_0_3/create_views.sql [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+            WHEN aud.general_information ->> 'multiple_eins_covered' = 'false' THEN 'No'
```
Comment (jadudm) [inline backend/dissemination/api/api_v1_0_3/create_views.sql:260]:
```suggestion
            WHEN aud.general_information ->> 'multiple_eins_covered' = 'false' THEN 'N'
```

**Review by jadudm — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**